disref/process.py

```diff
@@ -100,23 +106,72 @@ def create_reference(self, resource, block=True):
         :returns: The created Reference object
         """
 
-        self.client.hset(self.registry_key, resource, 1)
+        self.add_to_registry(resource)
         return Reference(self, resource, block)
 
+    def add_to_registry(self, resource, registry_key=None):
+        """
+        Adds a particular resource key to a process registry
+
+        :param str resource: An identifier for the resource.
+        :param str registry_key: Optional. The registry to which the
+            resource should be added.  If not specific, the resource
+            will be added to this process.
+
+        """
+        if registry_key is None:
+            registry_key = self.registry_key
+
+        return self.client.hset(self.registry_key, resource, 1)
+
+    def remove_from_registry(self, resource, registry_key=None):
+        """
+        Removes a particular resource key from a process registry
+
+        :param str resource: An identifier for the resource.
+        :param str registry_key: Optional. The registry from which the
+            resource should be removed.  If not specific, the resource
+            will be removed from this process.
+
+        """
+        if registry_key is None:
+            registry_key = self.registry_key
+
+        return self.client.hdel(self.registry_key, resource)
+
+    def get_registry(self, registry_key=None):
+        """
+        Returns a list of all items in a process' registry, excluding any
+        resources automatically created by each process.  As these are added,
+        they should be added to the REGISTRY_EXCLUSIONS below.
+
+        :param str registry_key: Optional. The registry for which to return
+            the registry.  If not specified, returns this process's registry.
+
+        """
+
+        REGISTRY_EXCLUSIONS = [self.heartbeat_hash_name]
```

Comment:
That was just an attempt to future-proof things a bit in the event that we wanted to exclude other references in the future.  But you're right, at the moment it doesn't serve any purpose.